pkg/workflow/workflow.go

```diff
@@ -156,6 +171,59 @@ func (w *workflow) Trace() error {
 	return recorder.With(w.cli, w.app).Save("", data).Limit(10).Error()
 }
 
+func (w *workflow) Cleanup(ctx monitorContext.Context) error {
+	ctxName := wfContext.GenerateStoreName(w.app.Name)
+	ctxCM := &corev1.ConfigMap{}
+	if err := w.cli.Get(ctx, client.ObjectKey{
+		Namespace: w.app.Namespace,
+		Name:      ctxName,
+	}, ctxCM); err != nil {
+		ctx.Error(err, "failed to get workflow context", "application", w.app.Name, "config map", ctxName)
```

Comment:
if not exist, it should not be an error

Reply:
In the cleanup case, yes, but this config map should exist since it's for workflow context.

Reply:
you could probably delete before and has a network issue. In this case, you should just return without error